src/shared/bootspec.c

```
@@ -1455,52 +1459,78 @@ static void print_addon(
                 _cleanup_strv_free_ char **ts = NULL;
 
                 t = strv_join(e->options, " ");
                 if (!t)
                         return log_oom();
 
-                ts = strv_split_newlines(t);
-                if (!ts)
+                if (indent_embedded_newlines(t, &options))
                         return log_oom();
 
-                t2 = strv_join(ts, "\n              ");
+                printf("      options: %s\n", options);
+                t2 = strdup(options);
                 if (!t2)
                         return log_oom();
-
-                printf("  ukiCmdline: %s\n", t2);
-                final_cmdline = TAKE_PTR(t2);
         }
 
-        FOREACH_ARRAY(addon, global_arr->items, global_arr->count) {
-                print_addon(addon, "globalAddon");
-                if (!strextend(&final_cmdline, " ", addon->cmdline))
+        FOREACH_ARRAY(addon, global_arr->items, global_arr->n_items) {
+                print_addon(addon, "global-addon");
+                if (!strextend(&t2, " ", addon->cmdline))
                         return log_oom();
         }
 
-        FOREACH_ARRAY(addon, e->local_addons.items, e->local_addons.count) {
+        FOREACH_ARRAY(addon, e->local_addons.items, e->local_addons.n_items) {
                 /* Add space at the beginning of addon_str to align it correctly */
-                print_addon(addon, " localAddon");
-                if (!strextend(&final_cmdline, " ", addon->cmdline))
+                print_addon(addon, " local-addon");
+                if (!strextend(&t2, " ", addon->cmdline))
                         return log_oom();
         }
 
-        if (final_cmdline)
-                printf(" finalCmdline: %s\n", final_cmdline);
+        /* Don't print the combined cmdline if it's same as options. */
+        if (streq_ptr(t2, options))
+                return 0;
+
+        if (indent_embedded_newlines(t2, &combined_cmdline))
+                return log_oom();
```

indent_embedded_newlines() already logged about all error conditions, hence please just propagate the error code here, do not log again. we generally want that every error is logged exactly once above LOG_DEBUG